packages/toolpad-core/src/dateFormat.tsx

```diff
@@ -0,0 +1,165 @@
+import { MenuItem, Stack, SxProps, TextField, styled } from '@mui/material';
+import * as React from 'react';
+
+export interface DateFormatPreset {
+  label?: string;
+}
+
+export type DateStyle = Intl.DateTimeFormatOptions['dateStyle'];
+export type TimeStyle = Intl.DateTimeFormatOptions['timeStyle'];
+
+const DEFAULT_DATE_STYLE: DateStyle = 'short';
+const DEFAULT_TIME_STYLE: TimeStyle = 'short';
+
+export const DATE_STYLES = new Map<DateStyle, DateFormatPreset>([
+  ['short', { label: 'Short' }],
+  ['medium', { label: 'Medium' }],
+  ['long', { label: 'Long' }],
+  ['full', { label: 'Full' }],
+]);
+
+export const TIME_STYLES = new Map<TimeStyle, DateFormatPreset>([
+  ['short', { label: 'Short' }],
+  ['medium', { label: 'Medium' }],
+  ['long', { label: 'Long' }],
+  ['full', { label: 'Full' }],
+]);
+
+const DATE_FORMATS = new Map<DateStyle, Intl.DateTimeFormat>(
+  (['short', 'medium', 'long', 'full'] as const).map((dateStyle) => [
+    dateStyle,
+    new Intl.DateTimeFormat(undefined, { dateStyle }),
+  ]),
+);
+
+const TIME_FORMATS = new Map<DateStyle, Intl.DateTimeFormat>(
+  (['short', 'medium', 'long', 'full'] as const).map((timeStyle) => [
+    timeStyle,
+    new Intl.DateTimeFormat(undefined, { timeStyle }),
+  ]),
+);
+
+const DEMO_DATE = new Date();
```

Comment:
Could also be the current time? Anyway not important at all :D

Reply:
I want a time that always demonstrates whether it's 12h or 24h clock

Reply:
I've added a comment